components/style/servo/url.rs

```diff
@@ -126,3 +127,25 @@ impl ToCss for SpecifiedUrl {
         dest.write_str(")")
     }
 }
+
+impl ToComputedValue for SpecifiedUrl {
+    type ComputedValue = ComputedUrl;
+
+    fn to_computed_value(&self, _: &Context) -> Self::ComputedValue {
+        match self.resolved {
+            Some(ref url) => ComputedUrl::Valid(url.clone()),
+            None => ComputedUrl::Invalid(Arc::new("about:invalid".to_string())),
```

Comment:
Sounds like a better idea.
I'll research other browsers this afternoon, I'll be in the Mozilla offices here in Berlin and I can probably ask directly :D